auction.sympl

```diff
@@ -33,95 +40,324 @@ schema Auction:
     creator: str
     closed: bool
 
-# Returns all the auctions in our storage
+# Admins are the only users allowed to:
+# 1. Create the Premium auction channel
+# 2. Select another admin to replace themselves (one serving admin at a time)
+# 3. Add/remove members in the Premium channel
+# The very first time a user is chosen to be an admin, that permission is given automatically.
+# Only one admin exists and has admin privileges at a time.
+schema Admin:
+    id: KeyAlias
+
+# A channel for private auctions.
+schema PremiumChannel:
+    channel_id: ChannelName
+    members : List[KeyAlias]
+
+###################################
+# Auction API and Implementations #
+###################################
+
+# Returns all auctions in storage.
 @clientside
 def get_auctions() -> List[Auction]:
-    return _get_auctions()
+    return cvm.storage.query(AuctionStatic).values()
 
-@helper
-def _get_auctions() -> List[Auction]:
-    return [row.value for row in cvm.storage.query(AuctionStatic).execute() ]
+# Returns a specific auction (if it exists).
+@clientside
+def get_auction(id: Identifier, in_premium: bool=False) -> Optional[Auction]:
+    channel : ChannelName = "PUBLIC"
+    if in_premium:
+        premium = cvm.storage.get(PUBLIC, PremiumChannelStatic, Identifier('premium'))
+        if isinstance(premium, PremiumChannel):
+            channel = premium.channel_id
+        else:
+            cvm.error(f"{channel} is not a valid channel.")
+
+    return cvm.storage.get(channel, AuctionStatic, id)
 
-# Returns a specific auction (if it exists)
+# Creates an auction.
+# Defaults to the PUBLIC channel, but members of the premium channel may post to it by specifying its ChannelName.
 @clientside
-def get_auction(id: Identifier) -> Optional[Auction]:
-    return cvm.storage.get(PUBLIC, AuctionStatic, id)
+def create_auction(product_description: str, initial_price: int, in_premium: bool=False) -> None:
+    channel : ChannelName = "PUBLIC"
+    if in_premium:
+        premium = cvm.storage.get(PUBLIC, PremiumChannelStatic, Identifier('premium'))
+        if isinstance(premium, PremiumChannel):
+            channel = premium.channel_id
+        else:
+            cvm.error(f"{channel} is not a valid channel.")
+
+    with PostTxArgs(channel):
+        _create_auction(product_description, initial_price)
+
+@executable
+def _create_auction(product_description: str, initial_price: int) -> None:
+    id = cvm.generate_id('AUC')
+    creator : str = cvm.tx.key_alias
+    channel : ChannelName = cvm.tx.write_channel
+    auction = cvm.storage.get(PUBLIC, AuctionStatic, id)
```

Comment:
You're only checking if the auction exists in the PUBLIC channel, what if you've already created an auction in the premium channel?